neo4j/internal/bolt/bolt4_test.go

```diff
@@ -671,6 +672,42 @@ func TestBolt4(outer *testing.T) {
 		AssertStringEqual(t, "", bookmark)
 	})
 
+	outer.Run("Begin transaction lazy with syncMessages false", func(t *testing.T) {
+		bolt, cleanup := connectToServer(t, func(srv *bolt4server) {
+			srv.accept(4)
+			srv.waitForTxBegin()
+			srv.sendFailureMsg("code", "Driver didn't lazily send begin")
+		})
+		defer cleanup()
+		defer bolt.Close(context.Background())
+
+		id, err := bolt.TxBegin(context.Background(),
+			idb.TxConfig{Mode: idb.ReadMode, Bookmarks: []string{"bm1"}}, false)
+
+		assertBoltState(t, bolt4_tx, bolt)
+		AssertNoError(t, err)
+		AssertNotDeepEquals(t, id, 0)
```

Comment:
What does this assert? Does `0` have a special semantic?

Reply:
It could be removed as the `AssertNoError` will fail first, but all of the failure branches in the method return `0` for the id